texar/utils/utils.py

```diff
@@ -605,6 +608,37 @@ def get_instance_kwargs(kwargs: Kwargs, hparams: ParamDict) -> Kwargs:
     return kwargs_
 
 
+def get_output_size(instance_: Optional[Module]) -> Optional[int]:
+    r"""Return the final dimension of output size from :attr:`instance_`.
+
+    If type of :attr:`instance_` is among the common types, the final
+    dimension of output size will be computed.
+
+    Args:
+        instance_: A :class:`~torch.nn.Module` instance from
+            which to compute output size.
+
+    :returns:
+        int (optional): The final dimension of the output size.
+    """
+    types = {
```

Comment:
Let's move this dictionary outside the method to prevent it from being constructed every time the method is called.